R/figures.R

```diff
@@ -30,10 +30,23 @@ get_plot_forecasted_counts <- function(draws,
                                        forecast_date,
                                        count_type = "hospital admissions",
                                        n_draws_to_plot = 100) {
-  sampled_draws <- sample(1:max(draws$draw), n_draws_to_plot)
+  n_draws_available <- max(draws$draw)
+  if (n_draws_available < n_draws_to_plot) {
+    stop(
+      sprintf(
+        "The number of draws to plot (%i) should be less or equal to ",
+        n_draws_to_plot
+      ),
+      sprintf(
+        "the number of draws in the data (%i).",
+        n_draws_available
+      )
+    )
+  }
+
```

Comment:
Suggest adding a test to ensure the correct df was passed into the function e.g.:


```suggestion

data_type <- draws |> dplyr::distinct("name") |> dplyr::pull()
if(data_type != "predicted_counts"){
    cli::cli_abort("Data passed to plotting function does not contain the expected variable type, `predicted_counts`")
}
```suggestion

Reply:
same for the other plotting functions

Reply:
I was adding this bit but realized this is not going to work as we don't have the column "name" anymore.